vite.config.js

```diff
@@ -1,32 +1,46 @@
 import { defineConfig } from 'vite'
+import { createHtmlPlugin } from 'vite-plugin-html'
 import path from 'path'
 import pluginRewriteAll from 'vite-plugin-rewrite-all'
 import svgLoader from 'vite-svg-loader'
 import vue from '@vitejs/plugin-vue'
 
 // https://vitejs.dev/config/
-export default defineConfig({
-  base: './',
-  // For testing, you may set the base path to a less conventional value to resemble a real user environment more closely. Annoyingly, this currently **requires** the trailing slash to be added in the URL. Navigating to http://localhost:8080/dev/gui will **not** work; http://localhost:8080/dev/gui/ will.
-  // Issue: https://github.com/vitejs/vite/issues/9236
-  // Fix in vite v4: https://github.com/vitejs/vite/blob/main/packages/vite/CHANGELOG.md#400-alpha1-2022-11-12
-  // base: '/dev/gui/',
-  server: {
-    port: 8080,
-  },
-  plugins: [
-    // Workaround for a bug with Vite serving a 404 page for routes that contain a “.” character.
-    // Issue: https://github.com/vitejs/vite/issues/2415.
-    pluginRewriteAll(),
-    vue(),
-    svgLoader(),
-  ],
-  resolve: {
-    alias: {
+export default defineConfig(function ({ mode }) {
+  return {
+    base: './',
+    server: {
+      port: 8080,
+    },
+    plugins: [
       /**
-       * Used to import files using, for example, '@/api/kumaApi'.
+       * Workaround for a bug with Vite serving a 404 page for routes that contain a “.” character.
+       *
+       * Issue: https://github.com/vitejs/vite/issues/2415.
        */
-      '@': path.resolve('./src'),
+      pluginRewriteAll(),
+      vue(),
+      svgLoader(),
+      createHtmlPlugin({
+        inject: {
+          data: {
+            /**
+             * Adds the appropriate GUI base path placeholder to the index.html file. It is going to be replaced using server-side templating when serving the GUI application in production.
+             */
+            baseGuiPath: mode === 'production' ? '{{.BaseGuiPath}}' : '/gui',
+            // **TIP**: Change this value to test various GUI base path scenarios.
```

Comment:
What are your thoughts on switching the `{{.}}` in an out here instead of dealing with it inside the application? I think we mentioned this idea elsewhere also. Feels like you could then reference `{baseGuiPath: '/gui'}.baseGuiPath` somehow here also.

```javascript
const config = {
  baseGuiPath: '/gui',
  ...
}
...
baseGuiPath: mode === 'production' ? '{{.BaseGuiPath}}' : config.baseGuiPath,
config: mode === 'production' ? '{{.}}' : JSON.stringify(config)
```

That kinda thing ^, then we remove all references to the string `/gui` from the application code entirely, its all just config.

Reply:
Yeah I like that idea. I’ve changed this around now.